ansible_base/feature_flags/models/aap_flag.py

```diff
@@ -0,0 +1,67 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from ansible_base.lib.abstract_models.common import NamedCommonModel
+from ansible_base.resource_registry.fields import AnsibleResourceField
+
+
+def validate_feature_flag_name(value: str):
+    if not value.startswith('FEATURE_') or not value.endswith('_ENABLED'):
+        raise ValidationError(_("Feature flag names must follow the format of `FEATURE_<flag-name>_ENABLED`"))
+
+
+class AAPFlag(NamedCommonModel):
+    class Meta:
+        app_label = "dab_feature_flags"
+        unique_together = ("name", "condition")
+
+    def __str__(self):
+        return "{name} condition {condition} is set to " "{value}{required}".format(
+            name=self.name,
+            condition=self.condition,
+            value=self.value,
+            required=" (required)" if self.required else "",
+        )
+
+    resource = AnsibleResourceField(primary_key_field="id")
+
+    name = models.CharField(
+        max_length=64,
+        null=False,
+        help_text=_("The name of the feature flag. Must follow the format of FEATURE_<flag-name>_ENABLED."),
+        validators=[validate_feature_flag_name],
+        blank=False,
+    )
+    ui_name = models.CharField(max_length=64, null=False, blank=False, help_text=_("The pretty name to display in the application User Interface"))
+    condition = models.CharField(max_length=64, default="boolean", help_text=_("Used to specify a condition, which if met, will enable the feature flag."))
+    value = models.CharField(
+        max_length=127,
+        default="False",
+        help_text=_("The value used to evaluate the conditional specified."),
+    )
+    required = models.BooleanField(
+        default=False,
+        help_text=_("If multiple conditions are required to be met to enable a feature flag, 'required' can be used to specify the necessary conditionals."),
+    )
+    support_level = models.CharField(
+        max_length=25,
+        null=False,
+        help_text=_("The support criteria for the feature flag. Must be one of (DEVELOPER_PREVIEW or TECHNOLOGY_PREVIEW)."),
+        choices=(('DEVELOPER_PREVIEW', 'Developer Preview'), ('TECHNOLOGY_PREVIEW', 'Technology Preview')),
+        blank=False,
+    )
+    visibility = models.BooleanField(
+        default=False,
+        help_text=_("The visibility of the feature flag. If false, flag is hidden."),
+    )
+    toggle_type = models.CharField(
+        max_length=20,
+        null=False,
+        choices=[('install-time', 'install-time'), ('run-time', 'run-time')],
+        default='run-time',
+        help_text=_("Details whether a flag is toggle-able at run-time or install-time. (Default: 'run-time')."),
+    )
```

Comment:
I looked in the proposal and I don't see any mention of this. I would like to get better clarity on how this will be used. The help text isn't actually doing it for me.

The default is incomparable with most currently-declared flags in `feature_flags.yml`.

By this, I mean, let's get together with UI developers who may work on it, like @nixocio, @PabloHiro on API design.

I just really want to get clarity on the meaning of this field, potential invalid states, and things like that. If we talk about this in advance, I do believe that is likely to prevent unnecessary migration work or bugs later on.